catalog/task/collect-extra-data/0.2/collect-extra-data.yaml

```diff
@@ -0,0 +1,101 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: collect-extra-data
+  labels:
+    app.kubernetes.io/version: "0.2"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task to collect extra data from release resources and optionally save CRs to a workspace
+  params:
+    - name: release
+      type: string
+      description: The namespaced name of the Release
+    - name: releasePlan
+      type: string
+      description: The namespaced name of the ReleasePlan
+    - name: releasePlanAdmission
+      type: string
+      description: The namespaced name of the ReleasePlanAdmission
+    - name: snapshot
+      type: string
+      description: The namespaced name of the Snapshot
+    - name: saveRelease
+      type: string
+      description: Save the Release CR as a json in the data workspace
+      default: true
+    - name: saveReleasePlan
+      type: string
+      description: Save the ReleasePlan CR as a json in the data workspace
+      default: true
+    - name: saveReleasePlanAdmission
+      type: string
+      description: Save the ReleasePlanAdmission CR as a json in the data workspace
+      default: true
+    - name: saveSnapshot
+      type: string
+      description: Save the Snapshot CR spec as a json in the data workspace
+      default: true
+  workspaces:
+    - name: data
+      description: Workspace to save the CR jsons to
+  results:
+    - name: extraData
+      description: |
+        Result of merging the `extraData` field of all the release resources
+  steps:
+    - name: collect-extra-data
+      image: quay.io/hacbs-release/release-utils:f486f350df43974706fc484a13bfeefc2c8fa95a
+      env:
+        - name: "RELEASE"
+          value: '$(params.release)'
+        - name: "RELEASE_PLAN"
+          value: '$(params.releasePlan)'
+        - name: "RELEASE_PLAN_ADMISSION"
+          value: '$(params.releasePlanAdmission)'
+        - name: "SNAPSHOT"
+          value: '$(params.snapshot)'
+      script: |
+        #!/usr/bin/env sh
+
+        if ! release_result=$(get-resource "release" "${RELEASE}" "{.spec.extraData}"); then
```

Comment:
Just to be sure, these checks will fail if the resource cannot be fetched, but not if the `extraData` field is missing, right? (as it's optional)

Could we maybe add an `echo` to say that the resource could not be fetched? Or will it be clear from the script output?

Reply:
ok, cool, thanks.